src/os/bluestore/AvlAllocator.cc

```diff
@@ -45,14 +45,16 @@ uint64_t AvlAllocator::_block_picker(const Tree& t,
     }
   }
   /*
-   * If we know we've searched the whole tree (*cursor == 0), give up.
-   * Otherwise, reset the cursor to the beginning and try again.
+   * If we reached end, start from beginning till cursor.
    */
-   if (*cursor == 0 || rs_start == t.begin()) {
-     return -1ULL;
-   }
```

Comment:
why is this part removed? i think it's still correct.

Reply:
+1

Reply:
It is removed, because if any of above conditions is true, then following loop is empty.
I removed it to simplify code.

Reply:
Just a remark here - when using best-fit mode (i.e. cursor=0 and lookup is performed using block size ordering) - one doesn't need to reiterate from the beginning - initial lookup has already provided the first suitable block - none are present before that position. Hence 
if(*cursor ==0) return -1;
might be useful.